src/hazelcore/extractions/ocaml/OCamlExtraction_Typ.re

```diff
@@ -0,0 +1,32 @@
+type t = string;
+
+let rec extract = (~t: HTyp.t): t =>
+  switch (t) {
+  | Hole => failwith("Typ: Hole type") //FIXME: remove Hole check in expression
+  | Int => "int"
+  | Float => "float"
+  | Bool => "bool"
+  // add parenthesis is necessary
+  // (int -> int) -> int != int -> int -> int
+  | Arrow(t1, t2) =>
+    "(" ++ extract(~t=t1) ++ " -> " ++ extract(~t=t2) ++ ")"
+  // use Polymorphic Variants to encode sum, i.e. 'Left of t1, 'Right of t2, and easier for injection
+  // the type is "[< `Left of t1 | `Right of t2]", < means no more than
+  // so the expression should be `Left e1, `Right e2 to match this design
+  | Sum(t1, t2) =>
+    "[< "
+    ++ "`Left of "
+    ++ extract(~t=t1)
+    ++ " | "
+    ++ "`Right of "
+    ++ extract(~t=t2)
+    ++ " ]"
+  | Prod(tl) => "(" ++ list_prod(~tl) ++ ")"
```

Comment:
careful with products, since internally we only use pairs but ocaml supports n-tuples. you might have to parenthesize in list_prod